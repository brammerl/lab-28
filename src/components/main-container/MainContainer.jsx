import React, { Component } from 'react'
import  UrlInput  from '../url-input/UrlInput';
import ResponseSection from '../response-section/ResponseSection';
import { fetchApi } from '../../utils/fetchApi';

export default class MainContainer extends Component {
  state = {
    url: '',
    method: '',
    body: '',
    display: { 'Hi': 'Please make a request'}
  }

  fetch = () => {
    const { url, method, body } = this.state;

    return fetchApi(url, method, body) 
      .then(res => this.setState({ display: res}))
  }

  handleSubmit = event => {
    event.preventDefault();
    this.fetch();
  }

  handleChange = ({target }) => {
    this.setState({ [target.name]: target.value})
  }

  render() {
    const { url, method, body, display } = this.state;
    return (
      <div>
        <UrlInput 
        url={url}
        body={body}
        onChange={this.handleChange}
        onSubmit={this.handleSubmit}/>
        <ResponseSection display={display}/>
      </div>
    )
  }
}
