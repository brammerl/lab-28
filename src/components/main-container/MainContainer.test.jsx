import React from 'react';
import { shallow } from 'enzyme';
import MainContainer from './MainContainer';

jest.mock('../../utils/fetchApi');

describe('Main container', () => {
  let wrapper;
  let main;

  beforeEach(() => {
    wrapper = shallow(<MainContainer/>);
    main = wrapper.instance();
  })

  it('renders the entire app', () => {
    expect(wrapper).toMatchSnapshot();
  })
  it('changes the url in state', () => {
    main.handleChange({
      target: {
        name: 'url',
        value: 'https://tester.com'
      }
    });

    expect(wrapper.state('url')).toEqual('https://tester.com');
  });

  it('changes the body in state', () => {
    main.handleChange({
      target: {
        name: 'body',
        value: 'testing body'
      }
    });

    expect(wrapper.state('body')).toEqual('testing body');
  })
})