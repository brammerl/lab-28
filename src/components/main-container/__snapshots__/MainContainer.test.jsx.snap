// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main container renders the entire app 1`] = `
<div>
  <UrlInput
    body=""
    onChange={[Function]}
    onSubmit={[Function]}
    url=""
  />
  <ResponseSection
    display={
      Object {
        "Hi": "Please make a request",
      }
    }
  />
</div>
`;
